# HG changeset patch
# User John W. Eaton <jwe@octave.org>
# Date 1324912667 18000
#      Mon Dec 26 10:17:47 2011 -0500
# Branch stable
# Node ID b8d9530e940e2e63793b9745e9ba7e6950e30436
# Parent  9627d5134a12fee1a9c46757310974575acbf64c
undo non-documentation changes in changeset 614505385171

* polyfit.m: Undo previous change.

diff --git a/scripts/polynomial/polyfit.m b/scripts/polynomial/polyfit.m
--- a/scripts/polynomial/polyfit.m
+++ b/scripts/polynomial/polyfit.m
@@ -76,33 +76,35 @@ function [p, s, mu] = polyfit (x, y, n)
   if (! size_equal (x, y))
     error ("polyfit: X and Y must be vectors of the same size");
   endif
 
   if (! (isscalar (n) && n >= 0 && ! isinf (n) && n == fix (n)))
     error ("polyfit: N must be a non-negative integer");
   endif
 
+  y_is_row_vector = (rows (y) == 1);
+
   ## Reshape x & y into column vectors.
   l = numel (x);
   x = x(:);
   y = y(:);
 
   ## Construct the Vandermonde matrix.
   v = vander (x, n+1);
 
   ## Solve by QR decomposition.
   [q, r, k] = qr (v, 0);
   p = r \ (q' * y);
   p(k) = p;
 
   if (nargout > 1)
     yf = v*p;
 
-    if (isrow (y))
+    if (y_is_row_vector)
       s.yf = yf.';
     else
       s.yf = yf;
     endif
 
     s.R = r;
     s.X = v;
     s.df = l - n - 1;
